src/libarena/lib.rs

```diff
@@ -486,9 +486,31 @@ impl DroplessArena {
         }
     }
 
+    #[inline]
+    unsafe fn write_from_iter<T, I: Iterator<Item = T>>(
+        &self,
+        mut iter: I,
+        len: usize,
+        mem: *mut T,
+    ) -> &mut [T] {
+        let mut i = 0;
+        // Use a manual loop since LLVM manages to optimize it better for
+        // slice iterators
+        loop {
+            let value = iter.next();
+            if !(i <= len && value.is_some()) {
```

Comment:
Shouldn't this be `i < len`? I'd find it easier to read if the condition was written as `i >= len || value.is_none()`.

Reply:
It should be `i < len` indeed.